Sources/SWBUniversalPlatform/TestEntryPointGenerationTaskAction.swift

```diff
@@ -66,18 +103,562 @@ class TestEntryPointGenerationTaskAction: TaskAction {
                         await Testing.__swiftPMEntryPoint() as Never
                     }
                     #endif
+                    if testingLibrary == "xctest" {
+                        #if !os(Windows) && \(options.enableExperimentalTestOutput)
+                        _ = SwiftPMXCTestObserver()
+                        #endif
+                        #if os(WASI)
+                        await XCTMain(__allDiscoveredTests()) as Never
+                        #else
+                        XCTMain(__allDiscoveredTests()) as Never
+                        #endif
+                    }
                 }
                 #endif
             }
-            """#)
+            """))
+
             return .succeeded
         } catch {
             outputDelegate.emitError("\(error)")
             return .failed
         }
     }
-}
 
-private struct Options: ParsableArguments {
-    @Option var output: Path
+    private struct Options: ParsableArguments {
+        @Option var output: Path
+        @Option var indexStoreLibraryPath: Path
+        @Option var linkerFilelist: [Path]
+        @Option var indexStore: [Path]
+        @Option var indexUnitBasePath: [Path]
+        @Flag var enableExperimentalTestOutput: Bool = false
+    }
+
+    private func discoveredTestsFragment(tests: [IndexStore.TestCaseClass]) -> String {
+        var fragment = ""
+        for moduleName in Set(tests.map { $0.module }).sorted() {
+            fragment += "@testable import \(moduleName)\n"
+        }
+        fragment += """
+        @available(*, deprecated, message: "Not actually deprecated. Marked as deprecated to allow inclusion of deprecated tests (which test deprecated functionality) without warnings")
+        public func __allDiscoveredTests() -> [XCTestCaseEntry] {
+            return [
+
+        """
+        for testClass in tests {
+            fragment += "        testCase(["
+            for method in testClass.testMethods {
+                let basename = method.name.hasSuffix("()") ? String(method.name.dropLast(2)) : method.name
+                fragment += "            (\"\(basename)\", \(testClass.name).\(basename))"
+            }
+            fragment += "        ]),\n"
+        }
+        fragment += """
+            ]
+        }
+        """
+        return fragment
+    }
+
+    private func testObservationFragment(testOutputPath: String) -> String {
+        """
+        #if !os(Windows) // Test observation is not supported on Windows
```

Comment:
Why? Do we have a GitHub issue for this?

Reply:
This is pulled in verbatim from SwiftPM